Do not remove custom errors in IR codegen when strip revert strings is requested (#16466)
ethereum/solidity: docs/using-the-compiler.rst

          // How to treat revert (and require) reason strings. Settings are
          // "default", "strip", "debug" and "verboseDebug".
          // "default" does not inject compiler-generated revert strings and keeps user-supplied ones.
          // "strip" removes all revert strings (if possible, i.e. if literals are used) keeping side-effects.

>           // NOTE: "strip" does not remove custom errors.

Reviewer: Actually, might be good to clarify that in `--help` as well.

>           // WARNING: Before Solidity 0.8.35, "strip" in the IR pipeline also removed custom errors.

Reviewer: That's not accurate. It would only remove them from `require()`. They were still preserved in the `revert` statement.

Reviewer: I'm not sure it's actually worth mentioning here though. It was a bug, not a design change. The entry in the changelog is probably enough.

>           // WARNING: Before Solidity 0.8.35, "strip" in the IR pipeline used to remove custom errors.

Reviewer 2: ```suggestion
          // WARNING: Before Solidity 0.8.35, "strip" in the IR pipeline also removed custom errors.
```

matter of taste
